services/ui-src/src/components/pages/Dashboard/DashboardTable.tsx

```diff
@@ -17,13 +24,25 @@ export const DashboardTable = ({ reports }: DashboardTableProps) => {
 
   const tableContent = {
     caption: "Quality Measure Reports",
-    headRow: ["Submission name", "Last edited", "Edited by", "Status", ""],
+    headRow: ["", "Submission name", "Last edited", "Edited by", "Status", ""],
   };
 
+  // pop off the edit button heading if the user is read only
+  if (readOnlyUser) {
+    tableContent.headRow.shift();
+  }
```

Comment:
Incoming waffle alert.

On the one hand, I don't like throwing away functions' return values. So I'd prefer `slice(1)` to `shift()` here.

But then we have to turn this into an assignment instead: `tc.hr = tc.hr.slice(1)`, which starts to feel awkward. Especially because we just declared the thing a second ago.

But maybe that's just revealing the underlying awkwardness of the whole situation, that we're creating something we might need to modify. So maybe we want to wrap the whole thing in a larger construct? Maybe along these lines???
```ts
const tableContent = {
  caption: "QMR",
  headRow: getHeadRow(readOnlyUser),
};
...

/** Return the dashboard table column headers, which differ for admin users */
const getHeadRow = (readOnlyUser: bool) => readOnlyUser
  ? ["Submission name", "Last edited", "Edited by", "Status", ""]
  : ["", "Submission name", "Last edited", "Edited by", "Status", ""];
```

But that's an awful lot of redundancy to include/exclude a single empty string. And maybe it'll be worth it someday, when the column headers change more? For example, maybe admins get an extra column for locking/archiving - that's what's happened in our other apps.

But then again maybe not. And surely we can trust our successors to be able to introduce such constructs as they become necessary, rather than us preemptively introducing them in the hopes that they'll be (understood and) used (well) later.

And now both you and I have spent more time writing/reading about these 3 lines of code than anyone is ever likely to in the future. So, do what you want, and I'll approve it.

Reply:
I like this way better than my hard to read implementation! I went ahead with your solution you provided. Thanks Ben!